sdk/benches/ed25519_instructions.rs

```diff
@@ -13,57 +13,78 @@ use {
     test::Bencher,
 };
 
-fn create_test_transaction(message_length: u16) -> Transaction {
-    let privkey = ed25519_dalek::Keypair::generate(&mut thread_rng());
-    let message_arr = vec![1u8; message_length as usize];
-    let instruction = new_ed25519_instruction(&privkey, &message_arr);
-    let mint_keypair = Keypair::new();
+// 5K transactions should be enough for benching loop
+const TX_COUNT: u16 = 5120;
 
-    Transaction::new_signed_with_payer(
-        &[instruction.clone()],
-        Some(&mint_keypair.pubkey()),
-        &[&mint_keypair],
-        Hash::default(),
-    )
+// prepare a bunch of unique txs
+fn create_test_transactions(message_length: u16) -> Vec<Transaction> {
+    (0..TX_COUNT)
+        .map(|_| {
+            let mut rng = thread_rng();
+            let privkey = ed25519_dalek::Keypair::generate(&mut rng);
+            let message: Vec<u8> = (0..message_length).map(|_| rng.gen_range(0, 255)).collect();
+            let instruction = new_ed25519_instruction(&privkey, &message);
+            let mint_keypair = Keypair::new();
+
+            Transaction::new_signed_with_payer(
+                &[instruction.clone()],
+                Some(&mint_keypair.pubkey()),
+                &[&mint_keypair],
+                Hash::default(),
+            )
+        })
+        .collect()
 }
 
 #[bench]
 fn bench_ed25519_len_032(b: &mut Bencher) {
-    let tx = create_test_transaction(32);
     let feature_set = FeatureSet::all_enabled();
-
+    let txs = create_test_transactions(32);
+    let mut n: usize = 0;
     b.iter(|| {
-        tx.verify_precompiles(&feature_set).unwrap();
+        txs[n % TX_COUNT as usize]
```

Comment:
Effectively the same, but maybe a little bit more "rusty":
```rust
let mut tx_iter = txs.iter().cycle();
b.iter(|| {
    tx_iter.next().unwrap().verify_precompiles(&feature_set).unwrap();
}
```

Reply:
neat! in 71babbe55f16e47136dabb28d0e6ef517fefdfb3